docs/getting_started/how_to_build_an_api.md

```diff
@@ -0,0 +1,215 @@
+
+[structmd]:# (pkg/apiserver/server.go HandlerMetadata Settings TimeoutSettings)
+**HandlerMetadata**
+
+HandlerMetadata stores the Path and Method of this Handler.
+
+| field       | type     | default     | description     |
+| :------------- | :----------: | :----------: | -----------: |
+| Method | string |  | Method is the route method of this Handler. |
+| Path | string |  | Path is the route path ot this handler. |
+
+**Settings**
+
+Settings structure for an API server.
+
+| field       | type     | default     | description     |
+| :------------- | :----------: | :----------: | -----------: |
+| Port | string | 8080 | Port the API listens to. |
+| Mode | string | release | Mode is either debug, release, test. |
+| Compression | CompressionSettings |  | Compression settings. |
+| Timeout | TimeoutSettings |  | Timeout settings. |
+
+**TimeoutSettings**
+
+TimeoutSettings configures IO timeouts.
+
+| field       | type     | default     | description     |
+| :------------- | :----------: | :----------: | -----------: |
+| Read |  | 60s | Read, write and idle timeouts. You need to give at least 1s as timeout.\nRead timeout is the maximum duration for reading the entire request, including the body. |
+| Write |  | 60s | Write timeout is the maximum duration before timing out writes of the response. |
+| Idle |  | 60s | Idle timeout is the maximum amount of time to wait for the next request when keep-alives are enabled |
+
+[structmd end]:#
+
+## Practical example: money-exchange application
+
+In `examples/getting_started/api` we can see the _money-exchange_ application. To run it:
+
+1. open a terminal, navigate to `examples/getting_started/api`, and type `$ go run main.go`
+2. open a second terminal, and type `curl localhost:8080/euro/10/GBP` or `curl localhost:8080/euro-at-date/10/USD/2021-01-03T00:00:00Z`
+
+To understand what _money-exchange_ does, let us start by looking at its _main_ function:
+
+[embedmd]:# (../../examples/getting_started/api/main.go /func main/ /\n}/)
+```go
+func main() {
+	app := application.New(
+		application.WithConfigFile("config.dist.yml", "yml"),
+		application.WithKernelSettingsFromConfig,
+		application.WithLoggerHandlersFromConfig)
+
+	app.Add("api", apiserver.New(definer.ApiDefiner))
+	app.Add("currency", currency.NewCurrencyModule())
+
+	app.Run()
+}
+```
+
+This application creates a kernel, using _config.dist.yml_ for its configuration, adds module _"currency"_ and _"api_ to it, then starts the kernel.
+
+These are the contents of _config.dist.yml_:
+```
+env: dev
+
+app_project: gosoline
+app_family: example
+app_name: money-exchange
+
+kernel:
+  killTimeout: 1s
+
+kvstore:
+  currency:
+    type: chain
+    in_memory:
+      max_size: 500000
+    application: money-exchange
+    elements: [inMemory]
+    ttl: 30m
+```
+
+The most interesting config is the currency key value store. This is defined as `inMemory` and serves as a local database.
```

Comment:
I would've been interested in how to configure an API server, kvstores are not really the topic here, right? But there is not even a single config setting for the API (because it is all using the defaults)

Reply:
Still, if I am reading the documentation about how I should build an API, I would at least expect some notion about how I can configure said API. How do I set the port would be the minimum set of config settings we should show here, even if it is redundant and already covered by the default